src/exportAssets.js

```diff
@@ -0,0 +1,55 @@
+const fs = require('uxp').storage.localFileSystem;
+const application = require("application");
+
+async function createDirectoryPath(topLevel, path) {
+  let folder = topLevel;
+
+  // Sequential execution to ensure that the toplevel is updated
+  for (const name of path) {
+    try {
+      folder = await folder.getEntry(name);
+    } catch(err) {
+      folder = await folder.createFolder(name);
+    }
+  }
+
+  return folder;
+}
+
+async function exportAssets(_selection, documentRoot) {
+  // TODO: confirm if marked_for export is enough or if every icon/illustration should be exported
+  //const assets = documentRoot.children.filter(artboard => artboard.name.startsWith('icons') || artboard.name.startsWith('illustrations'));
+  const assets = documentRoot.children.filter(artboard => artboard.markedForExport);
+
+  const folder = await fs.getFolder();
+  if (!folder) return;
+
+  /**
+   * Run sequential because application.createRenditions is not concurrent
+   * so we have to wait until the previous has resolved before starting a new one
+   */
+   for (const asset of assets) {
+    const path = asset.name.split('/');
```

Comment:
If you can use Node's `path` module you could go for `path.basename(asset.name)` here.
A basename is the lowest child (e.g file name) inside a tree of folders with its extension

Reply:
The individual parts are used to create the directory structure anyways